shard.go

```diff
@@ -34,68 +34,65 @@ type cacheShard struct {
 
 func (s *cacheShard) getWithInfo(key string, hashedKey uint64) (entry []byte, resp Response, err error) {
 	currentTime := uint64(s.clock.epoch())
-	wrappedEntry, err := s.getWrappedEntry(key, hashedKey)
-	if err == nil {
-		s.lock.RLock()
-		if entryKey := readKeyFromEntry(wrappedEntry); key != entryKey {
-			if s.isVerbose {
-				s.logger.Printf("Collision detected. Both %q and %q have the same hash %x", key, entryKey, hashedKey)
-			}
-			s.lock.RUnlock()
-			s.collision()
-			return entry, resp, ErrEntryNotFound
+	s.lock.RLock()
+	wrappedEntry, err := s.getWrappedEntry(hashedKey)
+	if err != nil {
+		s.lock.RUnlock()
+		return wrappedEntry, resp, err
+	}
+	if entryKey := readKeyFromEntry(wrappedEntry); key != entryKey {
+		if s.isVerbose {
+			s.logger.Printf("Collision detected. Both %q and %q have the same hash %x", key, entryKey, hashedKey)
 		}
+		s.lock.RUnlock()
+		s.collision()
+		return entry, resp, ErrEntryNotFound
+	}
 
-		oldestTimeStamp := readTimestampFromEntry(wrappedEntry)
-		if currentTime-oldestTimeStamp >= s.lifeWindow {
-			s.lock.RUnlock()
-			// @TODO: when Expired is non-default value return err as nil as the resp will have proper entry status
-			resp.EntryStatus = Expired
-			return entry, resp, ErrEntryIsDead
-		}
-		entry := readEntry(wrappedEntry)
+	oldestTimeStamp := readTimestampFromEntry(wrappedEntry)
+	if currentTime-oldestTimeStamp >= s.lifeWindow {
 		s.lock.RUnlock()
-		s.hit(hashedKey)
```

Comment:
@janisz I think we should still count it as hit, because the entry is still present in the cache after lifewindow expiry. If we do not count hit that would mis-lead to user who is looking at the stats for entries present in cache but not deleted.

Reply:
I did not change this part.This is line 61 of the new code

Reply:
👍🏽